### DISCUSSION TITLE
Non Uniform Strain Tensor

### DISCUSSION THREAD

**Discussion by vamatya** [FABLE-3DXRD/xrd_simulator#19] [General]
Hi, I'm trying to understand how to get diffraction patterns for a non-uniform strain applied to a given sample. For non-uniform strain, if I define something like: `[[0, 0, 0], 
     [0, 0.02 * x[1]/(2*sample_bounding_cylinder_radius), 0],
     [0, 0, 0]
     ]` for strain_tensor, I'd get peak broadening in the "angle-intensity" plot at about the center of original plot w/o strain. I'm interested to learn, if I can define a scenario where I can observe both peak broadening and peak shift at the same time; any suggestion on how to get that? Will translation (motion) of the sample help? Does volume of the sample w.r.t. the beam size play a role in this scenario?

**Comment by AxelHenningsson** [comment]
Some thoughts,

For a powdered sample, the sample size primarily determines the width of the diffraction rings, while the strain in the sample causes deformation of these rings. Different types of strain result in distinct deformations:

1. Volumetric strain predominantly expands or contracts the rings uniformly, leading to peak shifts without significant broadening.

2. Other strain states, such as uniaxial or shear strain, distort the circular rings into ellipses or other asymmetric shapes, which could result in both peak broadening and shifts in the diffraction pattern.

In your scenario, to observe both peak broadening and peak shift simultaneously, you might consider introducing a strain gradient (non-uniform strain) across the sample. This would superimpose varying local strain states, leading to a combination of ring deformation and broadening due to the distribution of strain.

**Reply by Leo-TKLiu** [reply]
Hello, I have follow-up questions. Applied with [[0, 0, 0],  [0, 0.02 * x[1]/(2*sample_bounding_cylinder_radius), 0], [0, 0, 0] ] for strain_tensor, the peak broadening is observed but no peak shift. Such strain gradient is symmetric to the sample, which leads to peak broadening due to volumetric expansion away from the beam center I guess. So here's my question: does the strain tensor change the lattice parameter or only the sample volume? If the strain is nonuniform but still symmetric to the sample body, is only peak broadening introduced or peak shift?

**Reply by vamatya** [reply]
Thanks for the insight Axel! To clarify the strain_tensor described above that @Leo-TKLiu and I explored, here is corresponding visualization: 
<img width="408" alt="Screenshot 2024-12-06 at 12 26 54 PM" src="https://github.com/user-attachments/assets/c48735ef-6dbc-43b8-a52a-5112badbb596">

Also, beam parameters used: 
-  "detector_distance": 1800000.0,
-  "detector_center_pixel_z": 1024,
-  "detector_center_pixel_y": 1024,
-  "pixel_side_length_z": 200,
-  "pixel_side_length_y": 200,
-  "number_of_detector_pixels_z": 2048,
-  "number_of_detector_pixels_y": 2048,
-  "wavelength": 0.1173,
-  "beam_side_length_z": 102400.0,
-  "beam_side_length_y": 102400.0,
-  "rotation_step": np.radians(2.0),
-  "rotation_axis": np.array([0., 0., 1.0])

This only results in broadening of the peak, and no shift.

**Reply by vamatya** [reply]
For Uni-axial strain, according to this [Uni-Axial Stress/Strain](https://csmbrannon.net/2012/08/02/distinction-between-uniaxial-stress-and-uniaxial-strain/), if I understood it correctly, we need  εA != 0 and εL == 0. So accordingly, we also tested with strain tensor as following: `[[0.02 * x[1]/(2*sample_bounding_cylinder_radius), 0, 0], 
     [0, 0, 0],
     [0, 0, 0]
     ]`.
This however does not show either peak broadening or shift. Corresponding visualization: <img width="546" alt="Screenshot 2024-12-06 at 12 50 35 PM" src="https://github.com/user-attachments/assets/e2eb7ab9-0620-43b6-ae89-3cefeb3f5b92">
